.github/actions/setup_env/action.yml

```diff
@@ -0,0 +1,42 @@
+name: "Set up python and conda environment"
+inputs:
+  cache-id:
+    description:
+      "Cache identifier used in the cache key. This should include any values
+      needed to uniquely identify the appropriate cache, such as python
+      version and operating system. The number of the current week in the
+      year, and a hash of all requirements files, are appended onto this
+      identifier to make the cache key. This way the cache will refresh each
+      week, or if the requirements change."
+    required: false
+    default: ""
+outputs:
+  cache-hit:
+    description: "True if a cache hit was found, false if not"
+    value: ${{ steps.condacache.outputs.cache-hit }}
+runs:
+  using: "composite"
+  steps:
+    # don't need to run setup-python because we'll always use the
+    # conda-managed python that'll be installed into this environment
+    - name: Setup conda environment
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        auto-update-conda: true
+        channels: conda-forge
+
+    # save week number to use in next step
+    # save CONDA_PREFIX to GITHUB_ENV so it's accessible outside of shell commands
+    - name: Set environment variables
+      run: |
+        echo "WEEK=$(date +%U)" >> $GITHUB_ENV
+        echo "CONDA_PREFIX=$CONDA_PREFIX" >> $GITHUB_ENV
+      shell: bash -l {0}
+
+    # NOTE the post step that saves the cache will only run if the job succeeds
+    - name: Restore conda environment cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.CONDA_PREFIX }}
+        key: ${{ inputs.cache-id }}-${{ env.WEEK }}-${{ hashFiles('**/requirements*.txt') }}
+      id: condacache
```

Comment:
General understanding question. Is the `id: condacache` here related and necessary to match the `ids` found in the workflow configs?

Reply:
Indirectly, yes. The `id` here is just used within this file. Then there's another `id` with the same name assigned to the step that calls the custom action.

Basically the custom action is a wrapper around the `actions/cache` action. The `id`s and `output`s are there to propagate the `cache-hit` output from the `actions/cache` up to the custom action so that it's available at the workflow level.

Trying to summarize another way:
1. the workflow step with id `condacache` runs the custom action
2. the custom action step with id `condacache` runs the `actions/cache` action 
3. the `actions/cache` action sets an output `cache-hit` on the custom action step
4. the custom action sets an output `cache-hit` on the workflow step that called it (using the value of `cache-hit` read from the custom action step)
5. future workflow steps can read the `cache-hit` output from the workflow step with id `condacache`